### PR TITLE
Added method to find a package according to a pattern

### PR DESCRIPTION
**Pull request by mchf**
## Problem

The feature was not supported. Some packages do not have "default" placeholder but uses versions in name instead (php8), so simple patterns comes handy.

- https://bugzilla.suse.com/show_bug.cgi?id=1200016

## Solution

implemented simple wrapper which uses y2packager/resolvable

## Testing

- *Tested manually*

### CODE REVIEW

File: library/packages/src/modules/Package.rb [hunk 1]
```diff
@@ -221,6 +222,18 @@ def reset
       @removed_packages.clear
     end
 
+    # Tries to find a package according to the pattern
+    #
+    # @param pattern [String] a pattern to match, no escaping done
+    # @return list of matching package names
+    def by_pattern(pattern = "")
+      init_packager
+      Y2Packager::Resolvable.find(provides_regexp: "^#{pattern}$")
```
Comment (mvidner) [inline library/packages/src/modules/Package.rb:231]: Maybe a comment noting that [Y2Packager::Resolvable.find](https://www.rubydoc.info/github/yast/yast-yast2/master/Y2Packager/Resolvable.find) takes POSIX EREs as opposed to Ruby Regexp below?

Also, its docs says to specify the used attrs for performance reasons
```suggestion
      Y2Packager::Resolvable.find(provides_regexp: "^#{pattern}$", [:name])
```

File: library/packages/src/modules/Package.rb [hunk 2]
```diff
@@ -221,6 +222,18 @@ def reset
...
+        .select{|p| p.name =~ /\A#{pattern}\z/ }
```
Comment (mvidner) [inline library/packages/src/modules/Package.rb:232]: `find` has filtered the results, why do we need to do it again? Add a comment?

File: library/packages/src/modules/Package.rb [hunk 3]
```diff
@@ -221,6 +222,18 @@ def reset
       @removed_packages.clear
     end
 
+    # Tries to find a package according to the pattern
+    #
+    # @param pattern [String] a pattern to match, no escaping done
+    # @return list of matching package names
+    def by_pattern(pattern = "")
```
Comment (mvidner) [inline library/packages/src/modules/Package.rb:229]: providing a default value for `by_pattern` IMHO only encourages unreadable call sites

File: library/packages/src/modules/Package.rb (hunk 1 above)
Comment (mchf) [reply library/packages/src/modules/Package.rb:231]: thanks for the tip

**Review by mvidner — APPROVED** [review]

**Pull request merged**